### PR TITLE
[Next.js] Remove Apollo dependency for GraphQL

### PR DESCRIPTION
**Pull request by illiakovalenko**
## Description
- `graphql-let` library requires to use `apollo-client` if we are using `react-apollo` plugin. I replaced it by lightweight `typed-document` library, it's only one feature which we needed.
- removed GraphQLClientFactory
- Used GraphQLRequestClient

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update (non-breaking change; modified files are limited to the `/docs` directory)

## Checklist:

- [ ] I have read the Contributing guide.
- [ ] My code follows the code style of this project.
- [ ] My code/comments/docs fully adhere to the Code of Conduct.
- [ ] My change is a code change and it requires an update to the documentation.
- [ ] My change is a documentation change and it requires an update to the navigation.

### CODE REVIEW

File: packages/sitecore-jss/src/graphql-request-client.ts [hunk 1]
```diff
@@ -14,7 +15,10 @@ export class GraphQLRequestClient {
    * @param {string} query graphql query
```
Comment (ambrauer) [inline packages/sitecore-jss/src/graphql-request-client.ts:15]: Update the param to reflect new allowed `DocumentNode`. Funny, I made the [same change in my branch](https://github.com/Sitecore/jss/commit/cc56d97c16cfad3a0e6f04e9d0e5732bdb8f7202#diff-ce9b582943d66dd42dbe58b3b6fe05f0c970ea184d8d41eddd5b93529b2906eaR23) 😄 
```suggestion
   * @param {string | DocumentNode} query graphql query
```

**Review by ambrauer — CHANGES_REQUESTED** [review]
👍 This turned out really nice - much cleaner in the sample app I think. Just the one update to the jsdoc required, otherwise looks great.

**Comment by nickwesselman** [comment]
Look at all those dependencies floating away... agree, this looks really nice.

File: packages/sitecore-jss/src/graphql-request-client.ts (hunk 1 above)
Comment (illiakovalenko) [reply packages/sitecore-jss/src/graphql-request-client.ts:15]: Eagle eye 👍  I added

**Pull request merged**